src/v2/manifest/mod.rs

```diff
@@ -141,6 +105,63 @@ impl Client {
         }
     }
 
+    async fn build_url(&self, name: &str, reference: &str) -> Result<Url> {
+        let ep = format!(
+            "{}/v2/{}/manifests/{}",
+            self.base_url.clone(),
+            name,
+            reference
+        );
+        match reqwest::Url::parse(&ep) {
+            Ok(url) => Ok(url),
+            Err(e) => {
+                return Err(Error::from(format!(
```

Comment:
If you want to make this a bit more compact, I think it is simply a `.map_err(|e| format!(...).into())`